File: ReactMeals_WebApi/Models/Token.cs
namespace ReactMeals_WebApi.Models;

public class Token
{
    public Token() { }
    public Token(string tokenValue, string tokenType, DateTime expiryDate)
    {
        TokenValue = tokenValue;
        TokenType = tokenType;
        ExpiryDate = expiryDate;
    }
    public int TokenId { get; set; }
    //base64 encoded
    public string TokenValue { get; set; }
    //"M_API" used for now only (Auth0 Management API token)
    public string TokenType { get; set; }
    public DateTime ExpiryDate { get; set; }
}

File: src/main/java/gr/jimmys/jimmysfoodzilla/models/Token.java
package gr.jimmys.jimmysfoodzilla.models;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "Tokens")
public class Token {
    public static String MANAGEMENT_API = "M_API";

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "Id")
    private int Id;

    //base64 encoded
    @Column(name = "TokenValue", columnDefinition = "VARCHAR(MAX)")
    private String tokenValue;

    //"M_API" used for now only (Auth0 Management API token)
    @Column(name = "TokenType", columnDefinition = "VARCHAR(MAX)")
    private String tokenType;

    @Column(name = "ExpiryDate")
    private LocalDateTime expiryDate;
}
